File: Old_Files_Not_relevant/w88try3/app/src/main/java/com/example/w88try3/MainActivity.java
package com.example.w88try3;

import static com.example.w88try3.R.id.TextView4;
import android.widget.TextView;
import android.os.AsyncTask;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;
import java.io.BufferedReader;

import java.io.OutputStream;
import java.net.HttpURLConnection;
import java.net.URL;
import java.io.InputStreamReader;

import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.NavigationUI;
import androidx.navigation.ui.AppBarConfiguration;



import android.view.Menu;
import android.view.MenuItem;

public class MainActivity extends AppCompatActivity {

    private TextView responseTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        responseTextView = findViewById(TextView4);
        Button myButton = findViewById(R.id.button);
        myButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                new SendHttpRequestTask().execute();
            }
        });
    }

    private class SendHttpRequestTask extends AsyncTask<Void, Void, String> {
        @Override
        protected String doInBackground(Void... params) {
            try {
                URL url = new URL("http://192.116.98.97:5000/api");
                HttpURLConnection conn = (HttpURLConnection) url.openConnection();
                conn.setRequestMethod("POST");
                conn.setRequestProperty("Content-Type", "application/json");
                conn.setDoOutput(true);

                String jsonInputString = "{\"hello\": \"tomer\"}";
                try (OutputStream os = conn.getOutputStream()) {
                    byte[] input = jsonInputString.getBytes("utf-8");
                    os.write(input, 0, input.length);
                }

                StringBuilder response = new StringBuilder();
                try (BufferedReader br = new BufferedReader(new InputStreamReader(conn.getInputStream(), "utf-8"))) {
                    int responseCode = conn.getResponseCode();
                    if (responseCode != HttpURLConnection.HTTP_OK) {
                        return "Error: Server responded with code " + responseCode;
                    }

                    String responseLine = null;
                    while ((responseLine = br.readLine()) != null) {
                        response.append(responseLine.trim());
                    }
                }

                conn.disconnect();
                return response.toString();
            } catch (Exception e) {
                e.printStackTrace();
                return "Error: " + e.getMessage();
            }
        }

        @Override
        protected void onPostExecute(String result) {
            responseTextView.setText(result);
        }
    }



    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

   /// @Override
   // public boolean onSupportNavigateUp() {
   //     NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment_content_main);
   //     return NavigationUI.navigateUp(navController, appBarConfiguration)
   //             || super.onSupportNavigateUp();
   // }
}

File: Old_Files_Not_relevant/W88takecare/app/src/main/java/com/example/w88takecare/MainActivity.kt
package com.example.w88takecare

import android.R
import android.os.AsyncTask
import android.view.View
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import java.net.HttpURLConnection
import java.net.URL
import java.io.OutputStream


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val myButton = findViewById<Button>(R.id.Button)
        myButton.setOnClickListener { SendHttpRequestTask().execute() }
    }

    private class SendHttpRequestTask :
        AsyncTask<Void?, Void?, Void?>() {
        protected override fun doInBackground(vararg params: Void): Void? {
            try {
                val url = URL("http://192.116.98.97:5000")
                val conn = url.openConnection() as HttpURLConnection
                conn.requestMethod = "POST"
                conn.setRequestProperty("Content-Type", "application/json")
                conn.doOutput = true
                val jsonInputString = "{\"hello\": \"tomer\"}"
                conn.outputStream.use { os ->
                    val input =
                        jsonInputString.toByteArray(charset("utf-8"))
                    os.write(input, 0, input.size)
                }

                // Read response, handle it or log it
                // ...
                conn.disconnect()
            } catch (e: Exception) {
                e.printStackTrace()
            }
            return null
        }
    }
}

